### PR TITLE
Add python client for FM API

### PR DESCRIPTION
**Pull request by thibaudio**
[ch63940]

### CODE REVIEW

**Comment by thibaudio** [comment]
I've updated the PR, but left the snapshots and getting the default vpc/subnet when creating networks out of the scope because the public api is lacking for now.

File: dataikuapi/fm/instancesettingstemplates.py [hunk 1]
```diff
@@ -0,0 +1,348 @@
...
+        """
+        Add setup actions
+
+        :param list setup_actions: List of :class:`dataikuapi.fm.instancesettingstemplates.FMSetupAction` to be played on an instance
+        :rtype: :class:`dataikuapi.fm.instancesettingstemplates.FMInstanceSettingsTemplateCreator`
+        """
+        self.data["setupActions"] = setup_actions
+        return self
+
+    def with_license(self, license):
```
Comment (FChataigner) [inline dataikuapi/fm/instancesettingstemplates.py:41]: missing docstring

**Review by FChataigner — CHANGES_REQUESTED** [review]
- not too fond of the methods `set_...()` that do a `save()` internally. I'd rather have the method return `self` so that you can chain a `save()` if you want
- no way to guess the image id for creation

File: dataikuapi/fmclient.py [hunk 2]
```diff
@@ -0,0 +1,375 @@
...
+        Do not create this class, instead use :class:`dataikuapi.FMClientAWS` or :class:`dataikuapi.FMClientAzure`
+        """
+        if self.cloud == None:
+            raise NotImplementedError(
+                "Do not use FMClient directly, instead use FMClientAWS or FMClientAzure"
+            )
+        self.api_key_id = api_key_id
+        self.api_key_secret = api_key_secret
+        self.host = host
+        self.__tenant_id = tenant_id
```
[the remark this answers is not in the document]
Comment (thibaudio) [reply dataikuapi/fmclient.py:53]: An old habit, `__` for private variables vs `_` for protected variables.
I can update it to a single underscore if you prefer to

**Review by FChataigner — APPROVED** [review]
LGTM

**Review by duckie — APPROVED** [review]

**Pull request merged**